--- origsrc/gio-sharp-2.22.3/configure.ac	2011-04-04 17:28:03.000000000 -0500
+++ src/gio-sharp-2.22.3/configure.ac	2011-05-17 18:02:34.575142500 -0500
@@ -36,16 +36,28 @@ PKG_CHECK_MODULES(GLIB, glib-2.0 >= 2.22
 
 CSC_FLAGS="-d:GIO_SHARP_2_22"
 
-if test `uname -s` = "Darwin"; then
-	LIB_PREFIX=
+AC_CANONICAL_HOST
+
+case "$host_os" in
+cygwin*)
+	LIB_PREFIX=cyg
+	LIB_MIDDLE=-
+	LIB_SUFFIX=.dll
+	;;
+darwin*)
+	LIB_PREFIX=lib
+	LIB_MIDDLE=.
 	LIB_SUFFIX=.dylib
-else
-	LIB_PREFIX=.so
+	;;
+*)
+	LIB_PREFIX=lib
+	LIB_MIDDLE=.so.
 	LIB_SUFFIX=
-fi
+esac
 
 AC_SUBST(CSC_FLAGS)
 AC_SUBST(LIB_PREFIX)
+AC_SUBST(LIB_MIDDLE)
 AC_SUBST(LIB_SUFFIX)
 
 AC_OUTPUT(
--- origsrc/gio-sharp-2.22.3/gio/gio-sharp.dll.config.in	2011-04-01 20:11:19.000000000 -0500
+++ src/gio-sharp-2.22.3/gio/gio-sharp.dll.config.in	2011-05-17 17:59:46.003500800 -0500
@@ -1,6 +1,6 @@
 <configuration>
-  <dllmap dll="libgio-2.0-0.dll" target="libgio-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libglib-2.0-0.dll" target="libglib-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libgthread-2.0-0.dll" target="libgthread-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="libgio-2.0-0.dll" target="@LIB_PREFIX@gio-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libglib-2.0-0.dll" target="@LIB_PREFIX@glib-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@LIB_PREFIX@gobject-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libgthread-2.0-0.dll" target="@LIB_PREFIX@gthread-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
 </configuration>
